File: src/easy/ToGoatLatin.swift
func toGoatLatin(_ sentence: String) -> String {
    if sentence.isEmpty {
        return ""
    }
    
    let vowels = "aeiouAEIOU"
    
    var words = sentence.split(separator: " ")
    
    for (i, word) in words.enumerated() {
        let suffix = String(repeating: "a", count: i + 1)
        
        if let firstCharacter = word.first {
            if vowels.contains(firstCharacter) {
                words[i] = "\(word)ma\(suffix)"
            } else {
                words[i] = "\(word[word.index(after: word.startIndex)...])\(firstCharacter)ma\(suffix)"
            }
        }
    }
    return words.joined(separator: " ")
}

File: lib/easy/to_goat_latin.dart
String toGoatLatin(String sentence) {
  
	List<String> words = sentence.split(" ");
  
	for (int i = 0; i < words.length; i++) {
    
		String suffix = "ma" + "a" * (i + 1);

    if ('aeiouAEIOU'.contains(words[i][0])) {
      words[i] = words[i] + suffix;
    } else {
      words[i] = words[i].substring(1) + words[i][0] + suffix;
    }
  }
  return words.join(" ");
}
